src/main/java/com/example/petbuddybackend/dto/user/ClientComplexInfoDTO.java

```diff
@@ -0,0 +1,12 @@
+package com.example.petbuddybackend.dto.user;
+
+import lombok.Builder;
+
+import java.util.Set;
+
+@Builder
+public record ClientComplexInfoDTO(
+        AccountDataDTO accountData,
+        Set<String> followingCaretakersEmails
+) {
+}
```

Comment:
> AccountDataDTO accountData

Zastanowiłbym się czy to nie przenieść do ClientDTO (`GET /api/client`) a w endpointach z followami tylko ten `Set<String> zwracać`. Szcególnie, że `AccountDataDTO` ma np. zdjęcie profilowe które trzeba odnawiać i imo to jest niepotrzebny overhead